### PR TITLE
Consolidated Resolution Changes

### PR DESCRIPTION
**Pull request by trieloff**
All changes suggested in adobe/helix-home#121

- [x] https://github.com/adobe/helix-dispatch/pull/247: `html` extension optional
- [x] https://github.com/adobe/helix-dispatch/pull/288: `README.html` no longer a directory index
- [x] https://github.com/adobe/helix-dispatch/issues/290: remove `default.html` support

Merge this one to rip off the band-aid.

### CODE REVIEW

File: src/fetchers.js [hunk 1]
```diff
@@ -110,25 +85,27 @@ function getDefault(url) {
  * @param {string[]} indices - array of indices.
  * @returns {PathInfo[]} An array of path info structures.
  */
-function getPathInfos(urlPath, mount, indices, resolveDefault = ((a) => a)) {
+function getPathInfos(urlPath, mount, indices) {
   // eslint-disable-next-line no-param-reassign
   urlPath = urlPath.replace(/\/+/, '/');
   // check if url has extension, and if not create array of directory indices.
   const urls = [];
+  const extensionlessurls = [];
```
Comment (tripodsan) [inline src/fetchers.js:93]: AFAICS, this array is never used ?

File: test/fetchers.test.js [hunk 2]
```diff
@@ -358,58 +353,24 @@ describe('testing error page promise resolver', () => {
...
 describe('testing path info resolution', () => {
   const tests = [
     {
-      url: '/',
+      url: '/hä',
       indices: ['index.html', 'readme.html'],
       mount: '',
       expected: [{
-        path: '/index.html',
+        path: '/hä/index.html',
```
Comment (tripodsan) [inline test/fetchers.test.js:363]: this is wrong. it should become `/hä.html`.
add also a test for `/hä/` which should become `/hä/index.html`

File: src/fetchers.js (hunk 1 above)
Comment (trieloff) [reply src/fetchers.js:93]: Indeed, this line https://github.com/adobe/helix-dispatch/pull/247/files#diff-fb1ce3759626b7da699b6918ce56f3a6R137 was merged away.

**Comment by trieloff** [comment]
There is no distinction between `/hä` and `/hä`. It will always try the directory index first, then `/hä.html`.

**Comment by trieloff** [comment]
If you want to build this distinction, feel free to do so.

File: test/fetchers.test.js [hunk 3]
```diff
@@ -356,7 +356,7 @@ describe('testing error page promise resolver', () => {
 describe('testing path info resolution', () => {
   const tests = [
     {
-      url: '/hä',
+      url: '/hä/',
```
Comment (trieloff) [inline test/fetchers.test.js:359]: Do you have a corresponding test without the trailing slash?

Comment (tripodsan) [reply test/fetchers.test.js:359]: yes:

https://github.com/adobe/helix-dispatch/blob/df9e22cc73db8ce4b8d7a82120e19eb38b87ae85/test/fetchers.test.js#L376-L387

Comment (trieloff) [reply test/fetchers.test.js:359]: Then we are good to go.

**Pull request merged**